File: seminar_2/task_3/patokova.victoria/task-3.py
def binary_search_sqrt(target):
    l = 0
    r = target

    while l <= r:
        middle = (l + r) // 2
        if middle * middle > target:
            r = middle - 1

        elif middle * middle < target:
            l = middle + 1

        else:
            return middle

    return r

File: seminar_2/task_3/churilkin.artem/task3.cpp
#include <iostream>

int binarySearchSqrt(int target) {
    int l = 0;
    int r = target;
    int middle = 0;
    while (l < r) {
        middle = (l + r) / 2;
        if (middle * middle < target) {
            l = middle + 1;
            continue;
        }
        if (middle * middle > target) {
            r = middle - 1;
            continue;
        }

        return middle;

    }
    
    return r;
}


int main() {
    int a = binarySearchSqrt(16); 
    if ( a == 4) std::cout << "ok";

    return 0;
}